components/org.wso2.carbon.identity.application.authenticator.cognito/src/main/java/org/wso2/carbon/identity/application/authenticator/cognito/CognitoOIDCAuthenticator.java

```diff
@@ -0,0 +1,318 @@
+        clientSecret.setDescription("Enter Cognito client secret value");
+        clientId.setDisplayOrder(2);
+        configProperties.add(clientSecret);
+
+        Property callbackUrl = new Property();
+        callbackUrl.setDisplayName("Callback Url");
+        callbackUrl.setName(CognitoOIDCAuthenticatorConstants.CALLBACK_URL);
+        callbackUrl.setDescription("Enter value corresponding to callback url.");
+        callbackUrl.setRequired(true);
+        callbackUrl.setDisplayOrder(3);
+        configProperties.add(callbackUrl);
+
+        Property authzEpUrl = new Property();
+        authzEpUrl.setName(CognitoOIDCAuthenticatorConstants.COGNITO_USER_POOL_DOMAIN);
+        authzEpUrl.setDisplayName("User Pool Domain");
+        authzEpUrl.setRequired(true);
+        authzEpUrl.setDescription("Enter Cognito user pool domain");
+        authzEpUrl.setDisplayOrder(4);
+        configProperties.add(authzEpUrl);
+
+        Property logoutRedirectUrl = new Property();
+        logoutRedirectUrl.setName(CognitoOIDCAuthenticatorConstants.LOGOUT_REDIRECT_URL);
+        logoutRedirectUrl.setDisplayName("Logout Redirect URL");
+        logoutRedirectUrl.setRequired(true);
+        logoutRedirectUrl.setDescription("Enter logout redirect url");
+        logoutRedirectUrl.setDisplayOrder(5);
+        configProperties.add(logoutRedirectUrl);
+
+        Property scope = new Property();
+        scope.setDisplayName("Additional Query Parameters");
+        scope.setName(CognitoOIDCAuthenticatorConstants.ADDITIONAL_QUERY_PARAMS);
+        scope.setDescription("Additional query parameters. e.g: paramName1=value1");
+        scope.setDisplayOrder(6);
+        configProperties.add(scope);
+
+        return configProperties;
+    }
+
+    @Override
+    protected void initiateLogoutRequest(HttpServletRequest request, HttpServletResponse response,
+                                         AuthenticationContext context) throws LogoutFailedException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Sending logout request to external IDP");
+        }
+        String clientId = context.getAuthenticatorProperties().get(CognitoOIDCAuthenticatorConstants.CLIENT_ID);
+        String logoutUrl = getLogoutUrl(context.getAuthenticatorProperties());
+        String logoutRedirectUrl = context.getAuthenticatorProperties().get(CognitoOIDCAuthenticatorConstants
+                .LOGOUT_REDIRECT_URL);
+
+        Map<String, String> parameters = new HashMap<>();
+
+        parameters.put(CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_CLIENT_ID, clientId);
+        parameters.put(CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_URI, logoutRedirectUrl);
+        FrameworkUtils.setCookie(request, response, CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_STATE_COOKIE,
+                context.getContextIdentifier(), null);
+        try {
+            response.sendRedirect(logoutUrl + "?" + getParamsString(parameters));
```

Comment:
logoutUrl + "?" + getParamsString(parameters)

Can we check whether there are available Util methods to achieve the same in FrameworkUtils class?